### DISCUSSION TITLE
PDF 1.7

### DISCUSSION THREAD

**Discussion by loust333** [gettalong/hexapdf#335] [Q&A] [answered]
Hello @gettalong,

I was trying to use HexaPDF to fill the following form: 

[declaration-exploitation-fr.pdf](https://github.com/user-attachments/files/17464232/declaration-exploitation-fr.pdf)

Is it compatible ? I was trying to get the list of all the fields but didn't get anything... 🤔

**Comment by gettalong** [comment, marked as answer]
That PDF does not contain a standard AcroForm interactive form but a (now deprecated) XFA form. XFA forms are currently not supported by HexaPDF and probably won't be in the near future (one reason: the XFA spec is 500 pages longer than the whole PDF 2.0 spec...).

**Reply by loust333** [reply]
Thanks, @gettalong. I saw in the documentation, but I was not sure if it was normal that the state used old documents. I have my answer 😅

**Reply by gettalong** [reply]
The documents are not exactly "old" but the technology is deprecated starting with PDF 2.0 (released in 2017). The thing is that the standard AcroForm can't yet do everything that the proprietary XFA does. So it is still used. And sometimes HexaPDF can process XFA forms if they are not actually dynamic.

However, in this case all the information is packed into the XFA dictionary.

**Reply by loust333** [reply]
What annoys me, is that they still use this technology and afterward everyone even non-tech users NEED to download the Acrobat Reader to be able of filling the PDF. 

I will try to solve using another official document that is readable but is only accepted per email. 😣

Thanks @gettalong.
